Notebook for all singler results

sjspielman opened the pull request:
### Purpose/implementation Section

#### Please link to the GitHub issue that this pull request addresses.

Closes #1172


#### What is the goal of this pull request?

For this PR, I ran `bash run-analysis.sh` (default settings!) through to generate our first round of SingleR results. This PR contains a notebook exploring the results. 


#### Briefly describe the general approach you took to achieve this goal.

This PR contains the following changes:

* Some reorganization
  * I moved the `jaccard-utils.R` file into `script/utils/` for broader usage throughout the module, and updated paths in the notebooks that use this file to match 
  * I ended up creating a subdir `exploratory-notebooks/singler-testing` to put all the notebooks and results that we've done so far, to keep this folder a bit less cluttered overall. I updated docs and paths to match this change.
* A palette
  * Inspired by the palette we have in `scpca-paper-figures`, I made a TSV with palette and harmonized (between consensus and nbatlas) cell type labels 
* The big notebook!
  * In this notebook, I use the merged file to look at all samples at once. Currently I am using a version of this file downloaded directly from the portal since the OpenScPCA version doesn't have consensus cell types in it (yet?). As such, I'm not currently running this notebook in CI.
  * The notebook includes a heatmap comparison to consensus, several views of the UMAP including comparison to consensus, and dotplots for both NBAtlas marker genes and consensus cell type marker genes

Here's a rendered version:
[singler-results.nb.html.zip](https://github.com/user-attachments/files/21186719/singler-results.nb.html.zip)

There are things the notebook currently does not do:
* I don't have too much text explanation in here yet because there's a lot to review here! So I thought it would be best to start review focused on just plots, and then for the next round (or in a separate PR - please let me know if you have a preference as reviewer!) I can add in the text. But very broadly speaking, the annotations are looking reasonable! 
* I don't currently export anything, but it would be useful to export a TSV of consensus & singler cell types to read in later for comparison with future scanvi results. I am thinking though it might be better to make a separate script to do this rather than having the notebook export it? Then the script can be run in CI, but the notebook would still not be.
* I have a couple big functions in the notebook; let me know if you want me to split these out into a utils file?



### Results

#### What is the name of your results bucket on S3?

Bucket: `researcher-654654257431-us-east-2`
Path: `s3://researcher-654654257431-us-east-2/cell-type-neuroblastoma-04/results/`

#### What types of results does your code produce (e.g., table, figure)?

All the SingleR results are there!

#### What is your summary of the results?

See the notebook :) 



### Provide directions for reviewers

#### What are the software and computational requirements needed to be able to run the code in this PR?

1. renv
2. Download the merged object for SCPCP000004 and store in `data/current/SCPCP000004` which is currently where the notebook expects it. 
  - But note, I'm not in love with that location!
  - I didn't want to put this in the `data/current/results` folder since it's not from OpenScPCA-nf.
  - I also haven't documented what to do about the fact that this file was manually downloaded yet...
  - Maybe it belongs in scratch, with docs in READMEs? 

#### Are there particularly areas you'd like reviewers to have a close look at?

Nothing beyond what has been mentioned


#### Is there anything that you want to discuss further?

N/A




### Author checklists

#### Analysis module and review

- [x] This analysis module [uses the analysis template and has the expected directory structure](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/).
- [x] The analysis module `README.md` has been updated to reflect code changes in this pull request.
- [x] The analytical code is documented and contains comments.
- [x] Any results and/or plots this code produces have been added to your S3 bucket for review.

#### Reproducibility checklist

- [ ] Code in this pull request has been added to the GitHub Action workflow that runs this module.
- [ ] The dependencies required to run the code in this pull request have been added to the analysis module `Dockerfile`.
- [ ] If applicable, the dependencies required to run the code in this pull request have been added to the analysis module conda `environment.yml` file.
- [x] If applicable, R package dependencies required to run the code in this pull request have been added to the analysis module `renv.lock` file.

allyhawkins commented:
> * I don't have too much text explanation in here yet because there's a lot to review here! So I thought it would be best to start review focused on just plots, and then for the next round (or in a separate PR - please let me know if you have a preference as reviewer!) I can add in the text. But very broadly speaking, the annotations are looking reasonable!

I took a preliminary look at the plots and honestly they look pretty good! I don't have a ton of comments or ideas for different plots outside of what you have here. One thing I noticed is that generally `SingleR` identifies more non-malignant cells than our consensus cell types do which doesn't surprise me at all. I think it could be helpful to dig into the non-malignant cells that get classified differently between the models. To do that I might make a dot plot looking at the validation markers but only include cells that are labeled by SingleR and not labeled by consensus? I want to see which one is more "correct"? Maybe instead of a dotplot it would make sense to look at a violin plot with the cells that agree plotted side by side with the cells that don't agree for each non-malignant cell type? 

> * I don't currently export anything, but it would be useful to export a TSV of consensus & singler cell types to read in later for comparison with future scanvi results. I am thinking though it might be better to make a separate script to do this rather than having the notebook export it? Then the script can be run in CI, but the notebook would still not be.

Yes I think we do want to export that TSV! I agree that it should be separate though and I would include that in a totally separate PR. 

> * I have a couple big functions in the notebook; let me know if you want me to split these out into a utils file?

Probably a good move? We may want to use similar functions when we get to looking at scANVI and SingleR together?

allyhawkins commented on analyses/cell-type-neuroblastoma-04/exploratory-notebooks/singler-results.Rmd:
```diff
@@ -0,0 +1,630 @@
+Next, we look at expression of consensus validation marker genes for each normal/harmonized cell type label.
+We compare expression across groups of cells:
+
+* `matching annotations`: Cells whose consensus annotation matches the SingleR annotation
+* `different annotations`: Cells whose consensus annotation differs from the SingleR annotation
+* `consensus not classified`: Cells where the consensus annotation is unknown
```
Sorry for the confusion with my initial comment, but I am specifically interested in the gene expression for cells that are classified as a normal cell type in SingleR and classified as unknown in consensus, so I would define these groups differently. I would divide like so: 
- cells that have a non-NE label in SingleR + classified as unknown by consensus
- all other cells that are either NE or not classified by SingleR + classified as unknown by consensus

Basically there are a group of cells for every cell type that `SingleR` classifies but consensus does not and I want to see if they do in fact have expression of the marker genes for that cell type. I would also use the NBAtlas marker genes here.

sjspielman commented on analyses/cell-type-neuroblastoma-04/exploratory-notebooks/singler-results.Rmd:
```diff
@@ -0,0 +1,630 @@
+  )
+
+# merged SCE file
+# OBTAINED FROM PORTAL DIRECTLY on 2025-07-09
```
Anyways, I got confused among all the release things, and consensus still aren't there but will be soon: https://github.com/AlexsLemonade/OpenScPCA-nf/issues/143

sjspielman commented:
Dotplot orders have been updated, and I updated the final plot as discussed, with some text updates throughout. As part of this I also added another little helper function to help prepare NBAtlas marker genes since I do that a couple times, and I'm sure it'll come back in another notebook one day!

Here's the current version:
[singler-results.nb.html.zip](https://github.com/user-attachments/files/21264581/singler-results.nb.html.zip)
